src/components/Table/EditableDraggableTable.js

```diff
@@ -146,28 +147,28 @@ class EditableDraggableTable extends React.Component {
 
   getActionButtons = (additionalData, isEditing) => {
     const renderConfig = this.getRenderConfig(additionalData);
-    const id = prefixedId(renderConfig.id, additionalData.rowKey);
-
-    const result =
-      isEditing && !renderConfig.visibleOnEdit ? null : (
-        <DropdownKebab className="row-actions" id={id} key={id} pullRight>
-          {renderConfig.actions.map((action, idx) => this.getActionButton(action, additionalData, prefixedId(idx, id)))}
-        </DropdownKebab>
-      );
+    let result;
+    if (renderConfig) {
+      const id = prefixedId(renderConfig.id, additionalData.rowKey);
+
+      result =
+        isEditing && !renderConfig.visibleOnEdit ? null : (
+          <DropdownKebab className="row-actions" id={id} key={id} pullRight>
+            {renderConfig.actions.map((action, idx) =>
+              this.getActionButton(action, additionalData, prefixedId(idx, id))
+            )}
+          </DropdownKebab>
+        );
+    }
 
     return <td className="editable">{result}</td>;
   };
 
   isDropdown = additionalData => get(this.getRenderConfig(additionalData), 'type') === 'dropdown';
 
   getRenderConfig = additionalData => {
-    const { renderConfigs } = additionalData.column;
-    const renderConfigIdx = additionalData.rowData.renderConfig;
-
-    if (renderConfigs && Number.isInteger(renderConfigIdx) && renderConfigIdx < renderConfigs.length) {
-      return renderConfigs[renderConfigIdx];
-    }
-    return null;
+    const { renderConfig } = additionalData.column;
+    return typeof renderConfig === 'function' ? renderConfig(additionalData.rowData) : renderConfig;
```

Comment:
can we also have the original declarative approach with this functional one?

Reply:
sure, lets do this in a follow-up